### DISCUSSION TITLE
Crazyflie 2.1 Brushless Motors - Only Three Motors Running, Fourth Motor Inactive

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1079] [Q&A] [answered]
Hi everybody, I am attempting to fly the `Crazyflie 2.1` with `Brushless Motors`. In `Crazyflie 2.1`, all four motor signals are at `PA1`, `PB11`, `PA15`, and `PB9`. I have connected these signals to my BLHeli ESC. Here is the platform config structure:

```c
static platformConfig_t configs[] = {
#ifdef CONFIG_SENSORS_BMI088_BMP388
  {
    .deviceType = "CF21",
    .deviceTypeName = "Crazyflie 2.1 Brushless",
    .sensorImplementation = SensorImplementation_bmi088_bmp388,
    .physicalLayoutAntennasAreClose = true,
    .motorMap = motorMapBoltBrushless
  },
#endif
};
```

You may wonder why I used `.motorMap = motorMapBoltBrushless`. Here is the reason:

```c
/**
 * Brushless motors mapped to the Bolt PWM outputs.
 */
const MotorPerifDef* motorMapBoltBrushless[NBR_OF_MOTORS] =
{
  &MOTORS_PA1_TIM2_CH2_BRUSHLESS_PP,
  &MOTORS_PB11_TIM2_CH4_BRUSHLESS_PP,
  &MOTORS_PA15_TIM2_CH1_BRUSHLESS_PP,
  &MOTORS_PB9_TIM4_CH4_BRUSHLESS_PP
};
```

However, the issue I'm facing is that only three motors are running, and `the fourth motor is not running`. I'm uncertain about the reason behind this. If you could assist me with this issue, it would be greatly appreciated!

**Comment by knmcguire** [comment, marked as answer]
Hi! Since you say that you are using the Crazyflie 2.1 for this, are you using the [Bigquad deck](https://www.bitcraze.io/products/bigquad-deck/) for this to interface with the brushless motors?

**Reply by amitx64** [reply]
@knmcguire, I'd like to clarify that I'm not utilizing the `BigQuad deck`. My current focus is on `mapping` the configuration of `Bolt Brushless motors` to the PWM outputs of the `Crazyflie 2.1`. Upon a thorough examination of the 'motors_def.h' file, I observed that one of the `Bolt` Motor GPIOs, specifically 'PB10,' is not assigned as a motor GPIO in the `Crazyflie 2.1`. Unfortunately, the reason for this discrepancy is unknown to me at this point. I suspect that due to this omission, one of the motors on my `Crazyflie 2.1` is not operational. This oversight is my responsibility, as I am currently in the experimental phase of testing various configurations. I appreciate your understanding as I work through these issues.

**Reply by tobbeanton** [reply]
That is a difference between Bolt 1.0 and Bolt 1.1
```
/**
 * Brushless motors mapped to the Bolt 1.1 PWM outputs.
 */
const MotorPerifDef* motorMapBolt11Brushless[NBR_OF_MOTORS] =
{
  &MOTORS_PA1_TIM2_CH2_BRUSHLESS_PP,
  &MOTORS_PB11_TIM2_CH4_BRUSHLESS_PP,
  &MOTORS_PA15_TIM2_CH1_BRUSHLESS_PP,
  &MOTORS_PB10_TIM2_CH3_BRUSHLESS_PP
};
```

**Reply by amitx64** [reply]
@tobbeanton Absolutely correct!.
